src/components/Personalization/decisionsExtractor.js

```diff
@@ -75,11 +77,35 @@ const extractDecisionsByScope = (decisions, scope) => {
   return { pageWideScopeDecisions, nonPageWideScopeDecisions };
 };
 
+const getGroupedDecisions = unprocessedDecisions => {
+  const decisionsGroupedByRedirectItemSchema = splitDecisions(
+    unprocessedDecisions,
+    REDIRECT_ITEM
+  );
+  const decisionsGroupedByDomActionSchema = splitDecisions(
+    decisionsGroupedByRedirectItemSchema.unmatchedDecisions,
+    DOM_ACTION
+  );
+
+  const {
+    pageWideScopeDecisions,
+    nonPageWideScopeDecisions
+  } = extractDecisionsByScope(
+    decisionsGroupedByDomActionSchema.matchedDecisions,
+    PAGE_WIDE_SCOPE
+  );
+
+  return {
+    redirectDecisions: decisionsGroupedByRedirectItemSchema.matchedDecisions,
+    pageWideScopeDecisions,
+    viewDecisions: nonPageWideScopeDecisions,
+    formBasedComposedDecisions:
+      decisionsGroupedByDomActionSchema.unmatchedDecisions
+  };
+};
+
 export default {
-  groupDecisionsBySchema({ decisions, schema }) {
-    return splitDecisions(decisions, schema);
-  },
-  groupDecisionsByScope({ decisions, scope }) {
-    return extractDecisionsByScope(decisions, scope);
+  groupDecisions(decisions) {
+    return getGroupedDecisions(decisions);
   }
```

Comment:
Since there is only one function that you are returning, you could change this filename to groupDecisions.js, and just export the one function.

Reply:
yes, it was already renamed to groupDecisions.